packages/plugin-core/src/commands/RefactorHierarchyV2.ts

```diff
@@ -32,27 +55,119 @@ export class RefactorHierarchyCommandV2 extends BasicCommand<
   CommandOutput
 > {
   key = DENDRON_COMMANDS.REFACTOR_HIERARCHY.key;
-  async gatherInputs(): Promise<CommandOpts | undefined> {
+
+  async promptScope(opts: {
+    initialValue: string;
+  }): Promise<NoteLookupProviderSuccessResp | undefined> {
+    const { initialValue } = opts;
+    const lcOpts: LookupControllerV3CreateOpts = {
+      nodeType: "note",
+      disableVaultSelection: true,
+      vaultSelectCanToggle: false,
+      extraButtons: [
+        Selection2ItemsBtn.create(false),
+        MultiSelectBtn.create(false),
+      ],
+    };
+    const controller = LookupControllerV3.create(lcOpts);
+    const provider = new NoteLookupProvider(this.key, {
+      allowNewNote: false,
+      noHidePickerOnAccept: false,
+    });
+    return new Promise((resolve) => {
+      NoteLookupProviderUtils.subscribe({
+        id: this.key,
+        controller,
+        logger: this.L,
+        onDone: (event: HistoryEvent) => {
+          const data = event.data as NoteLookupProviderSuccessResp;
+          if (data.cancel) {
+            resolve(undefined);
+          }
+          resolve(data);
+        },
+        onHide: () => {
+          resolve(undefined);
+        },
+      });
+      controller.show({
+        initialValue,
+        title: "Scope",
+        placeholder: "Query for scope. Press esc. to use all notes.",
+        provider,
+      });
+    });
+  }
+
+  async promptMatchText() {
+    const match = await VSCodeUtils.showInputBox({
+      prompt: "Enter match text. Leave blank to capture entire file name.",
+    });
+
+    if (match === undefined) {
+      // immediately return if user cancels.
+      return;
+    } else if (match.trim() === "") {
+      return "(.*)";
+    }
+    return match;
+  }
+
+  async promptReplaceText() {
+    let done = false;
     let replace: string | undefined;
+    do {
+      // eslint-disable-next-line no-await-in-loop
+      replace = await VSCodeUtils.showInputBox({
+        prompt: "Enter replace text.",
```

Comment:
Going along the lines of the previous comment:

Title: "Enter replace text."

Prompt: "This will replace the matched portion of the file name.  Regex is supported, including capture groups."

Placeholder: contents here depend on the scope set in stage 1.  If the user is trying to do the standard, current refactor op, we can probably leave this blank. Or maybe something like `new.hierarchy`.  If the user is trying the prefix refactor on selective note selection, then we can have a placeholder with the regex capture group syntax.

Reply:
I agree. I think it makes sense to do placeholders in the future for refactor-derivative commands that have very specific use cases, but for the vanilla refactor it's non-trivial to figure out a sensible placeholder for the replace stage